checkmate/checker/pipeline/core.py

```diff
@@ -0,0 +1,51 @@
+"""The core stage and pipeline objects."""
+
+import os.path
+from tempfile import NamedTemporaryFile
+
+from checkmate.exceptions import StageException
+
+
```

Comment:
A lot of comment and not much code here, but basically it's an interface for a callable in a working directory. And a pipeline that strings them all together. Doesn't make much sense on it's own, but take a look at the `URLHaus` class and it's quite obvious.